belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/api/InputValidationIssue.java

```diff
@@ -133,34 +143,84 @@ public List<Input<?>> getInputs() {
     }
 
     public void setInputs(List<Input<?>> inputs) {
-        verifyNoInNameValue();
+
         this.inputs.clear();
-        this.inputs.addAll(inputs);
+
+        if (inputs == null) {
+            return;
+        }
+
+        if (hasInNameValue()) {
+            throw new IllegalArgumentException(INPUTS_AND_IN_NAME_VALUE_ARE_MUTUALLY_EXCLUSIVE);
+        }
+
+        List<Input<?>> filteredInputs = inputs.stream().filter(Objects::nonNull).collect(Collectors.toList());
+
+        if (filteredInputs.size() == 1) {
+            throw new IllegalArgumentException(INPUTS_SETTER_ONE_ITEM);
+        }
+
+        this.inputs.addAll(filteredInputs);
     }
 
     public void setInputs(Input<?>... inputs) {
-        verifyNoInNameValue();
-        this.inputs.clear();
-        this.inputs.addAll(Arrays.asList(inputs));
+        if (inputs == null) {
+            return;
+        }
+        setInputs(Arrays.asList(inputs));
+    }
+
+    private void clearInNameValue() {
+        this.in = null;
+        this.name = null;
+        this.value = null;
     }
 
     public void addInput(Input<?> input) {
-        verifyNoInNameValue();
-        inputs.add(input);
+
+        if (input == null) {
+            return;
+        }
+
+        if (hasInNameValue()) {
+            this.inputs.add(new Input<>(in, name, value));
+            this.inputs.add(input);
+            clearInNameValue();
+        } else if (!inputs.isEmpty()) {
+            this.inputs.add(input);
+        } else {
+            in(input.getIn(), input.getName(), input.getValue());
+        }
     }
 
-    private void verifyNoInputs() {
-        if (!inputs.isEmpty()) {
-            throw new IllegalArgumentException(INPUTS_AND_IN_NAME_VALUE_ARE_MUTUALLY_EXCLUSIVE);
+    public void addInputs(List<Input<?>> input) {
+        if (input == null || input.isEmpty()) {
+            return;
         }
+        input.forEach(this::addInput);
     }
 
-    private void verifyNoInNameValue() {
-        if (in != null || name != null || value != null) {
+    public void addInput(Input<?>... input) {
+        if (input == null) {
+            return;
+        }
+        addInputs(Arrays.asList(input));
+    }
+
+    private void verifyNoInputs(Object valueToUpdate) {
+        if (valueToUpdate == null) {
```

Comment:
I wonder if this check with skip won't ignore undesired behavior. 

The method is called for any of the in/name/value setters. For in and name it doesn't make sense to be `null` to we'll ignore those. That leaves us with `setValue(null)`. If inputs wasn't empty we're attempting to combine 2 different concepts, even though the value is null. Shouldn't we continue to block that?

Reply:
To me, the goal of adding this ==null was about, if the setting parameter=null, then there is no reason to check if there is already anything in inputs list because we are not adding information. In my opinion, it would almost be a bug if we did this check in that case.

But, about this statement:

>  For in and name it doesn't make sense to be null to we'll ignore those

It would not make sense indeed but the code does not check if these are null but value is not for instance. So it would be interesting to add this validation. What do you guys think? Or it should be "free" like in the open-api contracts?

To make things simpler, we could just provide the in(in, name, value) method in which we would check first if the combination of provided info is acceptable. all the other methods would be private (in(), name(), etc).

Reply:
Ok to keep this check.

> It would not make sense indeed but the code does not check if these are null but value is not for instance. So it would be interesting to add this validation. What do you guys think?

I don't think it's possible to add a check for that since we don't have "build" method that indicates we're done with the initialization. When seeing a value but no in/name we don't know it they'll be set later on by the developer.

Individual "in()", "name()", ... methods make construction a bit more readable, especially if another method is called to determine their value.